platforms/Linux/centos/8/dev/patches/build-presets.patch

```diff
@@ -0,0 +1,23 @@
+
```

Comment:
Do we need this patch for CentOS 8?

Reply:
ideally I would like to get to zero patches. this one specifically does two things in build-presets:
1. uses the system ninja instead building it from source. we can also change the spec to download ninja and build it from source if preferable 
2. disables "test-installable-package", which I took from the fedora spec. not sure why it was needed there. @tachoknight?

Reply:
@tomerd I'm trying to get Fedora down to no patches as well. I have [this PR](https://github.com/apple/swift/pull/39463) which I called `buildbot_linux_fedora,no_test`, but maybe it would be better to make it more generic to RPM-based distros? The `test-installable-package` was the source of a lot of errors with Python, I'm trying a build with it enabled to see if it's still an issue.

Reply:
> ideally I would like to get to zero patches. this one specifically does two things in build-presets:
> 
> 1. uses the system ninja instead building it from source. we can also change the spec to download ninja and build it from source if preferable
I had to do this for CMake under CentOS 8 for awhile; I have an example if you'd like to see how to do it in a spec file [here](https://github.com/tachoknight/swift-lang-packaging-fedora/commit/45c989352718afbc0f66b5fb6434a51e36430756#diff-f2c19f70724bc875c60e718589aee604dbd71db0a2eb973a5dacd6b5e78a9908).
> 2. disables "test-installable-package", which I took from the fedora spec. not sure why it was needed there. @tachoknight?
I tested the latest Swift-5.5 snapshot with `test-installable-package` enabled and it worked, so this is no longer necessary to be removed.

Reply:
> uses the system ninja instead building it from source. we can also change the spec to download ninja and build it from source if preferable

@shahmishal  is there a reason not to use the system ninja when it is available?

Reply:
>> disables "test-installable-package", which I took from the fedora spec. not sure why it was needed there. @tachoknight?
> I tested the latest Swift-5.5 snapshot with test-installable-package enabled and it worked, so this is no longer necessary to be removed.

should we update the patch then?

Reply:
>uses the system ninja instead building it from source. we can also change the spec to download ninja and build it from source if preferable
@shahmishal is there a reason not to use the system ninja when it is available?

I dont see a problem in using the system ninja if it's available.

Reply:
>disables "test-installable-package", which I took from the fedora spec. not sure why it was needed there. @tachoknight?
I tested the latest Swift-5.5 snapshot with test-installable-package enabled and it worked, so this is no longer necessary to be removed.
@tachoknight should we update/remove the patch then?

I think we should remove the patch, and add support spec file to clone ninja.